### PR TITLE
mon: misc cleanups

### PR DESCRIPTION
**Pull request by xiexingguo**
Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

### CODE REVIEW

**Comment by xiexingguo** [comment]
@tchaikov Mind having a look? Thanks!

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -7732,7 +7732,7 @@ bool OSDMonitor::preprocess_pool_op(MonOpRequestRef op)
 
   if (!osdmap.get_pg_pool(m->pool)) {
     dout(10) << "attempt to operate on non-existent pool id " << m->pool << dendl;
-    _pool_op_reply(op, 0, osdmap.get_epoch());
+    _pool_op_reply(op, -ENOENT, osdmap.get_epoch());
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: could s/proper/consistent/ in commit message.

Comment (xiexingguo) [inline src/mon/OSDMonitor.cc]: :+1:

File: src/mon/LogMonitor.h [hunk 2]
```diff
@@ -206,6 +206,8 @@ class LogMonitor : public PaxosService,
       "mon_cluster_log_to_graylog",
       "mon_cluster_log_to_graylog_host",
       "mon_cluster_log_to_graylog_port",
+      "fsid",
```
Comment (tchaikov) [inline src/mon/LogMonitor.h]: > so LogMonitor::handle_conf_change() also reads from conf.host and pass the host read from conf to graylog.

@xiexingguo i cannot find the line where `host` or `fsid` is passed down to graylog in `LogMonitor`, could you point me to it? thanks.

**Comment by tchaikov** [comment]
modulo the change of "host" and "fsid", lgtm.

File: src/mon/LogMonitor.h (hunk 2 above)
Comment (xiexingguo) [inline src/mon/LogMonitor.h]: @tchaikov You are right. These two fields are not used and tracked. So we shall remove them from the `handle_conf_change` method.

**Comment by tchaikov** [comment]
lgtm

File: src/mon/OSDMonitor.cc (hunk 1 above)
Comment (xiexingguo) [inline src/mon/OSDMonitor.cc]: @tchaikov This change cause a lot of failures, such as:
http://qa-proxy.ceph.com/teuthology/yuriw-2016-08-16_17:06:30-rados-wip-yuri-testing_2016_08_15-distro-basic-smithi/368601/teuthology.log
http://qa-proxy.ceph.com/teuthology/yuriw-2016-08-16_17:06:30-rados-wip-yuri-testing_2016_08_15-distro-basic-smithi/368655/teuthology.log

**Comment by xiexingguo** [comment]
@yuriw Hi, yuri. The "pool existence checking" change in this PR cause a lot of failures, such as:


http://qa-proxy.ceph.com/teuthology/yuriw-2016-08-16_17:06:30-rados-wip-yuri-testing_2016_08_15-distro-basic-smithi/368601/teuthology.log
http://qa-proxy.ceph.com/teuthology/yuriw-2016-08-16_17:06:30-rados-wip-yuri-testing_2016_08_15-distro-basic-smithi/368655/teuthology.log


I dropped that change from this patch set and retested locally and passed:
http://www.daisycloud.org:9091/teuthology-2016-08-17_13:57:57-rados-wip-xxg-testing2---basic-plana/
http://www.daisycloud.org:9091/teuthology-2016-08-17_13:57:37-rados-wip-xxg-testing2---basic-plana/
http://www.daisycloud.org:9091/teuthology-2016-08-17_13:57:23-rados-wip-xxg-testing2---basic-plana/

Please reschedule in your next testing round. Sorry for the inconvenience.

**Comment by tchaikov** [comment]
@xiexingguo, could you investigate why ceph_test_rados_delete_pools_parallel failed?

**Comment by xiexingguo** [comment]
@tchaikov Previously there was a change I made in this patch to allow `OSDMonitor::preprocess_pool_op` return -ENOENT instead of 0 if the relevant pool was gone, which I think makes the pool deletion operation become non-idempotent and thus fail the `ceph_test_rados_delete_pools_parallel` script.

I drop that change and the failure is gone.

**Comment by tchaikov** [comment]
okay, got it!

**Comment by athanatos** [comment]
(NVM, 3ff81c7394c47104389fb7d658cb2fdafea0ba16 already got removed)

**Comment by tchaikov** [comment]
tested http://pulpito.ceph.com/kchai-2016-08-18_15:35:36-rados-wip-kefu-testing2---basic-mira/

all failures are known ones.

**Comment by xiexingguo** [comment]
@tchaikov Thanks, kefu, for the testing and merging.

**Pull request merged**